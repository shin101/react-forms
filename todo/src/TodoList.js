import React, {useState} from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid";

function TodoList(){

  const initialState = [
    { id:uuid(), name: 'go shopping' },
    { id:uuid(), name: 'have fun' }
  ]; 
  const [tasks, setTasks] = useState(initialState);
  const addTask = newTask => {
    setTasks(tasks => [...tasks, newTask]);
  };

  const remove = id => {
    setTasks(tasks => tasks.filter(task => task.id !== id));
  };
    
  return (
    <div>
    <h2>To Do List</h2>
    <NewTodoForm addTask={addTask} />
    <div>
    {tasks.map(({id, name}) => <Todo name={name} key={id} id={id} remove={remove} />)}
    </div>
  </div>) 
}



export default TodoList;