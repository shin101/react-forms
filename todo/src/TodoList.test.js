import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from './TodoList';



// smoke test
it("renders without crashing", function(){
  render(<TodoList />);
});

// snapshot test
it("matches snapshot", function (){
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
})

it("should add a new task", function(){
  const {queryByText, getByLabelText} = render(<TodoList />);
  expect(queryByText('hiking')).not.toBeInTheDocument();


  // if you add a new task will it correctly show on the page? 
  const input = getByLabelText("Tasks"); 
  const btn = queryByText("Add Task");
  fireEvent.change(input, {target: { value: 'go to the dentist'}});
  fireEvent.click(btn);
  expect(queryByText('go to the dentist')).toBeInTheDocument();
})

