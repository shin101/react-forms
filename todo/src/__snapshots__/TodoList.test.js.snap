// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      To Do List
    </h2>
    <form>
      <label
        for="name"
      >
        Tasks
      </label>
      <input
        id="name"
        name="name"
        placeholder="task name here"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
    <div>
      <div>
        <ul>
          <li>
            go shopping
          </li>
          <button>
            X
          </button>
        </ul>
      </div>
      <div>
        <ul>
          <li>
            have fun
          </li>
          <button>
            X
          </button>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
